bindings/src/querier.rs

```diff
@@ -110,11 +112,35 @@ impl<'a> ElysQuerier<'a> {
         );
         let request: QueryRequest<ElysQuery> = QueryRequest::Custom(query);
 
-        let resp: MarginOpenEstimationResponse = self.querier.query(&request)?;
+        let raw_resp: MarginOpenEstimationRawResponse = self.querier.query(&request)?;
+
+        let resp: MarginOpenEstimationResponse = MarginOpenEstimationResponse {
+            position: raw_resp.position,
+            leverage: Decimal::from_str(&raw_resp.leverage)
+                .map_or(Decimal::zero(), |leverage| leverage),
+            trading_asset: raw_resp.trading_asset,
+            collateral: raw_resp.collateral,
+            min_collateral: raw_resp.min_collateral,
+            valid_collateral: true, // Hard-coded as 'true' since the field is not found in the response
+            position_size: raw_resp.position_size,
+            swap_fee: Decimal::from_str(&raw_resp.swap_fee)
```

Comment:
does this return string type?

Reply:
So this return a result and if the conversion succeed it will return the value else it will return zero